src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -457,7 +525,15 @@ private unsafe BadHttpRequestException GetInvalidRequestException(RequestRejecti
             => KestrelBadHttpRequestException.GetException(
                 reason,
                 _showErrorDetails
-                    ? new Span<byte>(detail, length).GetAsciiStringEscaped(Constants.MaxExceptionDetailSize)
+                    ? new ReadOnlySpan<byte>(detail, length).GetAsciiStringEscaped(Constants.MaxExceptionDetailSize)
+                    : string.Empty);
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private BadHttpRequestException GetInvalidRequestException(RequestRejectionReason reason, ReadOnlySpan<byte> headerLine)
```

Comment:
Nit: why not have the unsafe version just call this?

Reply:
Done